.travis.yml

```diff
@@ -60,14 +12,90 @@ before_install:
   - chmod +x docker-compose
   - sudo mv docker-compose /usr/local/bin
 
-before_script:
-  - if [ "$CC" == "gcc" ]; then export CC=gcc-4.8; fi
-  - if [ "$CXX" == "g++" ]; then export CXX=g++-4.8; fi
-  - if [ "$CC" == "clang" ]; then export CC=clang-3.8; fi
-  - if [ "$CXX" == "clang++" ]; then export CXX=clang++-3.8; fi
-  - docker-compose build base
-  - if [ "$TARGET" == "android" ]; then docker-compose build java; fi
-  - docker-compose build ${TARGET}
-
-script:
-  - docker-compose run ${TARGET}
+jobs:
+  fast_finish: true
+  allow_failures:
+    - env: TARGET=sonar-scan
```

Comment:
I see. Well, we shouldn't be failing the build in case this is an external fork pull request and there're no variables set, because that's a legitimate case where it can fail - and we even have an "else" clause with `exit 0` for that, which never executes now.

At the same time we should fail it if variables are not set in a situation when it's a master branch push or an internal pull request. How about just moving the variable definition checks into a function and running the check for the `if` and `elif` clauses of the PR situation check? That will allow us to remove this "allowed failure" thing, which now is too broad IMO and lets both genuine as well as expected failures slip through. At the same time it doesn't require repo name hardcoding.